localstack/services/s3/s3_listener.py

```diff
@@ -1195,6 +1195,11 @@ def forward_request(self, method, path, data, headers):
                 return set_object_lock(bucket, data)
 
         path_orig_escaped = path_orig.replace('#', '%23')
+
+        # moto picks the region from the URL so adding the region param to the URL.
+        if method == 'PUT' and 'create-bucket' in headers.get('User-Agent'):
+            path_orig_escaped = path_orig_escaped + '.s3-{0}.amazonaws.com'.format(aws_stack.get_region())
```

Comment:
Thanks for this PR @usmangani1 . I think we should try to make this a bit more robust, depending on `User-Agent` may not be the best approach here (as the user agent string may significantly differ for different client SDKs).

Also, please note that we should generally support path-style as well as host-style bucket addressing. I think it would be beneficial to extend the unit test below with the following cases:
* path-style addressing, default region (`us-east-1`)
* host-style addressing, default region (`us-east-1`)
* host-style addressing, non-default region (`us-west-1`)

Regarding the logic around `path_orig_escaped` - assuming the original path is `/mybucket` (for path-style addressing), we would generate a value like `/mybucket.s3-us-west-1.amazonaws.com` for `path_orig_escaped`, is that right? So, the request we forward to the backed would look something like this - is that our intention?
```
PUT http://localhost:4566/mybucket.s3-us-west-1.amazonaws.com
```

Wondering if we should instead look into fixing/modifying the `Authorization` header (as this should generally be used to determine the target region).